src/DistributedMetadata/CatalogService.cpp

```diff
@@ -52,7 +52,10 @@ Int32 searchIntValueByRegex(const std::regex & pattern, const String & str)
     assert(m);
     (void)m;
 
-    return std::stoi(pattern_match.str(1));
+    if (pattern_match.size() > 0)
```

Comment:
we already assert at line 52. May i ask what kind of cases we are handling here ?

Reply:
When using tcp to create the DistributedMergeTree table, Just like the following statement does not assigned shard :

CREATE TABLE default.`abcc`　(　    `col_1` Nullable(Int32) DEFAULT 1 CODEC(ZSTD(1)),　    `col_2` Float32 DEFAULT 1.1,　    `col_3` DateTime DEFAULT now() CODEC(ZSTD(1))　)　ENGINE = DistributedMergeTree(1, 1, 0)　PARTITION BY toYYYYMM(col_3)　ORDER BY (toYYYYMMDD(col_3), (col_3, col_2))　TTL col_3 + toIntervalMonth(1)

Reply:
1. We don't support TCP to create distributed mergetree table now ?
2. if we do, we will need do the same way as HTTP does ? Each table definition needs have an associated `shard`, otherwise it won't work even we create the table successfully

Reply:
Have withdrawed changes